utils/mgmt_system/kubernetes.py

```diff
@@ -0,0 +1,181 @@
+from collections import namedtuple
+from mgmtsystem.base import ContainerMgmtSystemAPIBase
+from mgmtsystem.rest_client import ContainerClient
+
+"""
+Related yaml structures:
+
+[cfme_data]
+management_systems:
+    kubernetes:
+        name: My kubernetes
+        type: kubernetes
+        hostname: 10.12.13.14
+        port: 6443
+        credentials: kubernetes
+        authenticate: true
+        rest_protocol: https
+
+[credentials]
+kubernetes:
+    username: admin
+    password: secret
+    token: mytoken
+"""
+
+# Below are simplified representations of CFME k8s objects that should cover most of our needs
+Container = namedtuple('Container', ['name', 'cg_name', 'image'])
+Image = namedtuple('Image', ['name', 'tag', 'id'])
+# Images are equal if their ids match
+Image.__eq__ = lambda self, other: self.id == other.id
```

Comment:
Not that this would bother me much, but I'd prefer it to return `False` if eg. the types don't match so it does not raise exceptions when `.id` is not present in the `other`. Like when you write eg. `1 == []` in python.

```python
Image.__eq__ = lambda self, other: hasattr(other, "id") and self.id == other.id
```